pennychat/permissions.py

```diff
@@ -0,0 +1,10 @@
+from rest_framework.permissions import BasePermission
+
+
+class IsOwner(BasePermission):
+    """
+    Allows access only to the user who owns the entity.
+    """
+
+    def has_permission(self, request, view):
+        return bool(view.get_object().user_profile.user == request.user)
```

Comment:
I'm interested in understanding how this works. The logic looks right of course. But what is `get_object()` doing? This will bread on some views certainly, right? (those w/o a user_profile) Does this need a little more documentation?

Reply:
You're right, I should add documentation.